lib/Serialization/Serialization.cpp

```diff
@@ -579,10 +579,11 @@ void Serializer::writeHeader(const SerializationOptions &options) {
     ModuleName.emit(ScratchRecord, M->getName().str());
 
     SmallString<32> versionStringBuf;
+    auto version = M->getASTContext().LangOpts.EffectiveLanguageVersion;
     llvm::raw_svector_ostream versionString(versionStringBuf);
-    versionString << Version::getCurrentLanguageVersion();
+    versionString << version;
```

Comment:
Well, it does put the full version string _after_ the '/' in the header; I saw (and intended to preserve) the short bit at the beginning is used to detect format mismatches, but I thought that was the point here: that compiling in `-swift-version 3` will produce a version-3-compatible module, and that we're going to conditionalize any evolution of the module format to being "in later effective versions".

Perhaps I'm misunderstanding?